File: app/src/main/java/com/omarea/shell/units/FlymeUnit.java
package com.omarea.shell.units;

import com.omarea.shell.SysUtils;

import java.util.ArrayList;

/**
 * Created by Hello on 2017/11/01.
 */

public class FlymeUnit {
    public boolean StaticBlur() {
        ArrayList<String> commands = new ArrayList<String>() {{
            add("setprop persist.sys.static_blur_mode true");
        }};
        return SysUtils.executeRootCommand(commands);
    }

    public boolean DisableMtkLog() {
        ArrayList<String> commands = new ArrayList<String>() {{
            add("rm -rf /sdcard/mtklog");
            add("echo 0 > /sdcard/mtklog");
        }};
        return SysUtils.executeRootCommand(commands);
    }
}

File: app/src/main/java/com/omarea/shell/units/FlymeUnit.kt
package com.omarea.shell.units

import com.omarea.shared.CommonCmds
import com.omarea.shell.SuDo
import java.util.*

/**
 * Created by Hello on 2017/11/01.
 */

class FlymeUnit {
    fun staticBlur(): Boolean {
        val commands = object : ArrayList<String>() {
            init {
                add("setprop persist.sys.static_blur_mode true")
                add(CommonCmds.MountSystemRW)
                add("busybox sed 's/^persist.sys.static_blur_mode=.*/persist.sys.static_blur_mode=true/' /system/build.prop > /data/build.prop;")
                add("cp /data/build.prop /system/build.prop\n")
                add("rm /data/build.prop\n")
                add("chmod 0644 /system/build.prop\n")
                add(CommonCmds.Reboot)
            }
        }
        return SuDo.execCmdSync(commands)
    }
}
